packages/context/src/interception-proxy.ts

```diff
@@ -77,6 +94,7 @@ export class InterceptionHandler<T extends object> implements ProxyHandler<T> {
           target,
           propertyName,
           args,
+          {source: this.session && new ProxySource(this.session)},
```

Comment:
Should we use `??` operator here?

```suggestion
          {source: this.session ?? new ProxySource(this.session)},
```

Reply:
Will switch after #4213 is landed.

Reply:
Hmm, I see this error:
```
src/interception-proxy.ts:97:52 - error TS2345: Argument of type 'undefined'
is not assignable to parameter of type 'ResolutionSession'.

97           {source: this.session ?? new ProxySource(this.session)},
```

It seems that `tsc` does not use `??` as an assertion.

Reply:
> `? this.session ?? new ProxySource(this.session)`

My fault. The code above says "use `this.session` if it is set, otherwise create new ProxySource".  (It's a replacement for `||`, not `&&`).